### PR TITLE
Add Delete note functionality, and bug fixes

### LINKED ISSUE #116: Add ability to delete notes
Allow users to delete unwanted notes.

### LINKED ISSUE #123: Bug: Editing book removes all notes.
+ Change the return result of adding note command.

### PR DESCRIPTION
**Pull request by pangpuncake**
Closes #116 and Closes #123

### CODE REVIEW

**Comment by pangpuncake** [comment]
Main changes:
Add the delete note functionality.
Removed bug in edit command which causes loss of notes after editing.
Made the result from adding a new note to a book more concise for users to easily read.
Reformatted UG to keep a consistent format among features. 
Built a prefix table for users to better understand each prefix and their meanings before delving into the feature list.

File: src/main/java/seedu/bookmark/model/book/Note.java [hunk 1]
```diff
@@ -24,6 +24,7 @@
      * @param text A valid text string.
      */
     public Note(String title, String text) {
+        assert (title != null && text != null);
```
Comment (TanLeYang) [inline src/main/java/seedu/bookmark/model/book/Note.java:27]: is this redundant since there's a requireNonNull right after already? 😆

**Review by TanLeYang — COMMENTED** [review]
Nice job! LGTM except for one assertion which I thought looks a bit redundant

File: src/main/java/seedu/bookmark/model/book/Note.java (hunk 1 above)
Comment (pangpuncake) [reply src/main/java/seedu/bookmark/model/book/Note.java:27]: This was for the assertions requirement for the week haha! Should I remove it?

File: src/test/java/seedu/bookmark/logic/commands/DeleteNoteCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        assertTrue(outOfBoundIndex.getZeroBased() < model.getLibrary().getBookList().size());
+
+        DeleteNoteCommand deleteNoteCommand = new DeleteNoteCommand(outOfBoundIndex,
+               Index.fromOneBased(1));
+
+        assertCommandFailure(deleteNoteCommand, model, Messages.MESSAGE_INVALID_BOOK_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_validBookIndexInvalidNoteIndexFilteredList_failure() {
```
Comment (angrybunny123) [inline src/test/java/seedu/bookmark/logic/commands/DeleteNoteCommandTest.java:88]: Nice comprehensive testing Wai Kye!

**Review by angrybunny123 — COMMENTED** [review]
LGTM :)

**Review by pennhanlee — COMMENTED** [review]
LGTM!

**Pull request merged**